### ISSUE TITLE
Simple documentation of automatic optimization with learning rate scheduler

### ISSUE THREAD

**Issue by turian** [Lightning-AI/lightning#16564]
### 📚 Documentation

The documentation for the [learning rate scheduler](https://pytorch-lightning.readthedocs.io/en/stable/common/optimization.html) is unclear. It focuses on manual optimization, and automatic optimization is buried in a note without a code example. Adding, the documentation for `configure_optimizers` is not linked and that section has no examples of `configure_optimizers` usage.

This has caused confusion in the past, for example:
* https://lightning.ai/forums/t/lr-scheduler-with-interval-step-not-working-properly/732
* https://github.com/Lightning-AI/lightning/issues/9475
* https://github.com/Lightning-AI/lightning/issues/4456

I suggest the section begin with a simple code example with automatic optimization, explaining what will happen for all four cases of frequency and interval being defined or undefined. Alternately, that section linked above should say: "For a simple example of learning rate scheduling with automatic optimization, see the documentation for `[configure_optimizers](https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#configure-optimizers)`"

cc @borda

**Comment by pranith7** [comment]
Hey @Borda can i work on this?. Thank you

**Comment by Borda** [comment]
> Hey @Borda can i work on this?. Thank you

Sure, go ahead! and thank you :)

**Comment by noobjam** [comment]
Hi @Borda can i help with this ?

**Comment by Bhavay-2001** [comment]
Hi @Borda, is this issue open right now? I would like to work on this.
Thanks

**Comment by Bhavay-2001** [comment]
Hi @Borda, can you quickly guide me please on how should I proceed? Where to start looking from?
Thanks

**Comment by nlgranger** [comment]
Also the `manual_training = True` example does not mention that the scheduler must be stepped manually. One could be led to think this is done automatically elsewhere.

**Comment by TanvirTurja** [comment]
Hi @Borda, is this still available? I'd like to work on this.
